modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,568 @@
+
+    private static final int DEBUGGING_PORT = 9000;
+    private static final int MQTT_PORT = 1883;
+    private static final int CONTROL_CENTER_PORT = 8080;
+    @SuppressWarnings("OctalInteger")
+    private static final int MODE = 0777;
+    private static final @NotNull Pattern EXTENSION_ID_PATTERN = Pattern.compile("<id>(.+?)</id>");
+
+    private final @NotNull ConcurrentHashMap<String, CountDownLatch> containerOutputLatches = new ConcurrentHashMap<>();
+    private boolean controlCenterEnabled = false;
+    private boolean debugging = false;
+
+    private final @NotNull LinkedList<String> prepackagedExtensionsToRemove = new LinkedList<>();
+    private boolean removeAllPrepackagedExtensions = false;
+
+    private final @NotNull MultiLogMessageWaitStrategy waitStrategy = new MultiLogMessageWaitStrategy();
+
+    public HiveMQContainer(final @NotNull DockerImageName dockerImageName) {
+        super(dockerImageName);
+
+        dockerImageName.assertCompatibleWith(DEFAULT_HIVEMQ_CE_IMAGE_NAME, DEFAULT_HIVEMQ_EE_IMAGE_NAME);
+
+        addExposedPort(MQTT_PORT);
+
+        waitStrategy.withRegEx("(.*)Started HiveMQ in(.*)");
+        waitingFor(waitStrategy);
+
+        withLogConsumer(outputFrame -> {
+            final String utf8String = outputFrame.getUtf8String();
+            if (debugging && utf8String.startsWith("Listening for transport dt_socket at address:")) {
+                System.out.println("Listening for transport dt_socket at address: " + getMappedPort(DEBUGGING_PORT));
+            }
+            if (!containerOutputLatches.isEmpty()) {
+                containerOutputLatches.forEach((regEx, latch) -> {
+                    if (outputFrame.getUtf8String().matches("(?s)" + regEx)) {
+                        LOGGER.debug("Container Output '{}' matched RegEx '{}'", utf8String, regEx);
+                        latch.countDown();
+                    } else {
+                        LOGGER.debug("Container Output '{}' did not match RegEx '{}'", utf8String, regEx);
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    protected void configure() {
+        withCreateContainerCmdModifier(it -> it.withEntrypoint("/bin/sh"));
+
+        final String removeCommand;
+
+        if (removeAllPrepackagedExtensions) {
+            removeCommand = "rm -rf /opt/hivemq/extensions/** &&";
+        } else {
+            removeCommand = prepackagedExtensionsToRemove.stream()
+                .map(extensionId -> "rm -rf /opt/hivemq/extensions/" + extensionId + "&&")
+                .collect(Collectors.joining());
+        }
+        setCommand(
+            "-c",
+            removeCommand +
+                "cp -r '/opt/hivemq/temp-extensions/'* /opt/hivemq/extensions/ " +
+                "; rm -rf /opt/hivemq/temp-extensions/** " +
+                "&& chmod -R 777 /opt/hivemq/extensions " +
+                "&& /opt/docker-entrypoint.sh /opt/hivemq/bin/run.sh"
+        );
```

Comment:
not major, but... have you considered doing it only if needed? (either `removeAllPrepackagedExtensions` is set or `prepackagedExtensionsToRemove` isn't empty)

This would make it a bit more future proof, in case `/opt/hivemq/bin/run.sh` gets a new location for example